iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -383,13 +518,302 @@ public static void checkPhaseTapChangerRegulation(Validable validable, PhaseTapC
+     * If the voltage control is enabled voltage setpoint must be defined and valid.
+     */
+    public static void checkVoltageControlTargetQ(Validable validable,
+        double voltageSetpoint, double reactivePowerSetpoint, boolean voltageRegulatorOn) {
+        if (!voltageRegulatorOn && !validReactivePowerSetpoint(reactivePowerSetpoint)) {
+            throw createInvalidValueException(validable, reactivePowerSetpoint, REACTIVE_POWER_SETPOINT, VOLTAGE_REGULATOR_OFF);
+        }
+
+        checkVoltageRegulation(validable, voltageSetpoint, voltageRegulatorOn);
+    }
+
+    private static void checkVoltageRegulation(Validable validable, double voltageSetpoint, boolean voltageRegulatorOn) {
+        if (!voltageRegulatorOn) {
+            return;
+        }
+        if (!validVoltageSetpoint(voltageSetpoint)) {
+            throw createInvalidValueException(validable, voltageSetpoint, VOLTAGE_SETPOINT, "voltage regulator is on");
+        }
+    }
+
+    /**
+     * For validating discrete voltage controls.
+     * Regulating Terminal is forced to be validated.
+     */
+    public static void checkDiscreteVoltageControl(Validable validable, Terminal regulatingTerminal,
+        double voltageSetpoint, double targetDeadband, boolean voltageRegulatorOn, Network network) {
+        checkDiscreteVoltageControl(validable, regulatingTerminal, voltageSetpoint, targetDeadband, voltageRegulatorOn,
+            network, true);
+    }
+
+    /**
+    * For validating discrete voltage controls.
+    * User can select if regulating terminal should be validated.
+    * If the voltage regulation control is enabled voltage setpoint, target deadband and regulating terminal must be defined and valid.
+    */
+    public static void checkDiscreteVoltageControl(Validable validable, Terminal regulatingTerminal,
+        double voltageSetpoint, double targetDeadband, boolean voltageRegulatorOn, Network network,
+        boolean validateRegulatingTerminal) {
+        if (!voltageRegulatorOn) {
+            return;
+        }
+        if (validateRegulatingTerminal && !validRegulatingTerminal(regulatingTerminal, network)) {
+            throw new ValidationException(validable, REGULATING_TERMINAL_NOT_DEFINED);
+        }
+        if (!validVoltageSetpoint(voltageSetpoint)) {
+            throw createInvalidValueException(validable, voltageSetpoint, VOLTAGE_SETPOINT);
+        }
+        if (!validDeadband(targetDeadband)) {
+            throw createInvalidValueException(validable, targetDeadband, "target deadband");
+        }
+    }
+
+    /**
+    * For validating reactive power controls.
+    * If the control is enabled reactive power setpoint and regulating terminal must be defined and valid.
+    */
+    public static void checkReactivePowerControl(Validable validable, Terminal regulatingTerminal,
+        double reactivePowerSetpoint, boolean reactivePowerRegulatorOn, Network network) {
+        if (!reactivePowerRegulatorOn) {
+            return;
```

Comment:
Same remark as above

Reply:
Changed. When the control is disabled only valid values and the initial one (attribute is not set) are allowed.